src/app/venter/complaints-home/complaints-home.component.ts

```diff
@@ -0,0 +1,150 @@
+import { Component, OnInit } from '@angular/core';
+import { IComplaint } from '../../interfaces';
+import { DataService } from '../../data.service';
+import { API } from '../../../api';
+import { MatSnackBar } from '@angular/material/snack-bar';
+import * as moment from 'moment';
+
+@Component({
+  selector: 'app-complaints-home',
+  templateUrl: './complaints-home.component.html',
+  styleUrls: ['./complaints-home.component.css'],
+})
+export class ComplaintsHomeComponent implements OnInit {
+
+  public complaints = {} as IComplaint[];
+  public myComplaints = {} as IComplaint[];
+  public userid: string;
+  public complaintstatusColor: string[] = [];
+  public myComplaintstatusColor: string[] = [];
+  public reportedTime: string[] = [];
+  public myComplaintsFlag: boolean;
+  public complaintsFlag: boolean;
+
+  constructor(
+    public dataService: DataService,
+    public snackBar: MatSnackBar,
+  ) { }
+
+  ngOnInit() {
+    this.complaints = [];
+    this.myComplaints = [];
+    /* Get profile if the user is logged in */
+    if (this.dataService.isLoggedIn()) {
+      this.getUser();
```

Comment:
This is an async call, so you have no guarantee it is going to complete before proceeding. Everything happening after this that needs the user object needs to be inside the subscription

Reply:
I didn't quite get what you meant by this. I think I am subscribing to the result of GetFillCurrentUser() method in getuser(). Can you please explain?

Reply:
Since `GetFillCurrentUser` returns an observable, this call _could_ take (say) a minute to complete, in which case everything else would be done before that, and the user would still not be available. To avoid this, anything that needs the user should happen _inside_ the subscription, i.e. it should be guaranteed to complete after we have completed the async user call.